### PR TITLE
fix(RetrieveExternalData): amend code to allow for URLs for different…

### PR DESCRIPTION
**Pull request by lydiagray**
… envs

### Description
Amend PageAction to allow for different endpoints for each env

### Checklist
- [x ] Code compiles correctly
- [x ] Created tests for the new changes
- [x ] All tests passing
- [x ] Extended the README / documentation, if necessary

### CODE REVIEW

File: src/Helpers/PageHelpers/PageHelper.cs [hunk 1]
```diff
@@ -407,7 +408,13 @@ private void CheckRetrieveExternalDataAction(List<IAction> actions)
 
             actions.ForEach(action =>
             {
-                if (string.IsNullOrEmpty(action.Properties.URL))
+                var foundSlug = new PageActionSlug();
+                foreach (var slug in action.Properties.PageActionSlugs.Where(_ => _.Environment.ToLower().Equals(_environment.EnvironmentName.ToS3EnvPrefix().ToLower())))
```
Comment (lallysmbc) [inline src/Helpers/PageHelpers/PageHelper.cs:412]: foreach loop can be removed and replaced with FirstOrDefault.

var foundSlug = action.Properties.PageActionSlugs.FirstOrDefault(_ => _)

File: src/Services/RetrieveExternalDataService/RetrieveExternalDataService.cs [hunk 2]
```diff
@@ -40,10 +44,16 @@ public async Task Process(List<IAction> actions, FormSchema formSchema, string f
             foreach (var action in actions)
             {
                 var response = new HttpResponseMessage();
-                var entity = _actionHelper.GenerateUrl(action.Properties.URL, mappingData.FormAnswers);
+                var submitSlug = action.Properties.PageActionSlugs.FirstOrDefault(_ =>
+                    _.Environment.Equals(_environment.EnvironmentName.ToS3EnvPrefix()));
```
Comment (lallysmbc) [inline src/Services/RetrieveExternalDataService/RetrieveExternalDataService.cs:48]: could we add a toLower to this check on both, to remove any possibility of case issues.

File: tests/form-builder-tests-unit.csproj [hunk 3]
```diff
@@ -9,6 +9,7 @@
 
   <ItemGroup>
     <PackageReference Include="AWSSDK.S3" Version="3.3.104.41" />
+    <PackageReference Include="AWSSDK.SimpleEmail" Version="3.3.101.175" />
```
Comment (lallysmbc) [inline tests/form-builder-tests-unit.csproj:12]: is this package needed in unit test project or can be removed.

Comment (lydiagray) [reply tests/form-builder-tests-unit.csproj:12]: It is needed, the tests use the SimpleEmail.Model

**Pull request merged**